### PR TITLE
add a lot more modeling

### PR DESCRIPTION
**Pull request by kfinn**
introduce the concept of a membership pair

room channel now only notifies on new membership pairs
when a user subscribes to a room channel, create a membership for that user and also membership pairs between that membership and each existing one
notify everybody involved

membership pair channel receives offer, answer, and ice candidate events
offers, answers, and ice candidates saved to db, referencing membership pair entries
when a user subscribes to their membership pair channel, they receive any existing offers, answers, and ice candidates from the db

hoping this allows for easier client-side modeling

### CODE REVIEW

File: app/models/membership_pair.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    def must_have_exactly_two_membership_pair_entries
+        errors[:membership_pair_entries] << 'must have exactly two' unless membership_pair_entries.size == 2
+    end
+
+    def membership_pair_entry_for_user(user)
+        membership_pair_entries.find_by!(membership: user.memberships)
+    end
+
+    def membersihp_pair_entry_not_for_user(user)
```
Comment (samandmoore) [inline app/models/membership_pair.rb:14]: membership

File: app/models/membership_pair.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class MembershipPair < ApplicationRecord
+    has_many :membership_pair_entries
+
+    validate :must_have_exactly_two_membership_pair_entries
+
+    def must_have_exactly_two_membership_pair_entries
+        errors[:membership_pair_entries] << 'must have exactly two' unless membership_pair_entries.size == 2
```
Comment (samandmoore) [inline app/models/membership_pair.rb:7]: I thought this shovel syntax was deprecated on rails 6

**Comment by samandmoore** [comment]
These tables might grow v fast eventually lol so we can revisit later

File: app/models/membership_pair.rb (hunk 1 above)
Comment (kfinn) [reply app/models/membership_pair.rb:14]: whoops

File: app/models/membership_pair.rb (hunk 2 above)
Comment (kfinn) [reply app/models/membership_pair.rb:7]: whoops

**Comment by kfinn** [comment]
merging to deploy! 🙏

**Pull request merged**